go/grpcweb/wrapper.go

```diff
@@ -57,13 +76,29 @@ func WrapServer(server *grpc.Server, options ...Option) *WrappedGrpcServer {
 //
 // You can control the CORS behaviour using `With*` options in the WrapServer function.
 func (w *WrappedGrpcServer) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
+	if w.enableWebsockets && w.IsGrpcWebSocketRequest(req) {
+		if w.websocketOriginFunc(req) {
+			if !w.opts.corsForRegisteredEndpointsOnly || w.isRequestForRegisteredEndpoint(req) {
+				w.HandleGrpcWebsocketRequest(resp, req)
+				return
+			}
+		}
+		resp.WriteHeader(400)
```

Comment:
`StatusBadRequest`? Doesn't `StatusForbidden` (403) make more sense here? 403 instead of 401 because it's Forbidden from this Origin and not a matter of authentication.

Reply:
Agreed, 403 is better.